runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2443,18 +2432,52 @@ J9::Z::TreeEvaluator::instanceofEvaluator(TR::Node * node, TR::CodeGenerator * c
       }
    }
 
+/** \brief
+ *     Generates null test of \p objectReg for instanceof or checkcast[AndNULLCHK] \p node. In case a NULLCHK is
+ *     required this function will generate the sequence which throws the appropriate exception.
+ *
+ *  \param node
+ *     The instanceof, checkcast, or checkcastAndNULLCHK node.
+ *
+ *  \param cg
+ *     The code generator used to generate the instructions.
+ *
+ *  \param objectReg
+ *     The object to null test.
+ */
 static void
-generateNullChkSnippet(
-      TR::Node *node,
-      TR::CodeGenerator *cg)
+genInstanceOfOrCheckCastNullTest(TR::Node* node, TR::CodeGenerator* cg, TR::Register* objectReg)
    {
-   TR::Compilation *comp = cg->comp();
-   TR::LabelSymbol * snippetLabel = generateLabelSymbol(cg);
-   TR::S390BranchInstruction * brInstr = (TR::S390BranchInstruction*) generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BE, node, snippetLabel);
-   brInstr->setExceptBranchOp();
+   const bool isNullTestImplicit = node->getOpCodeValue() == TR::checkcastAndNULLCHK && cg->getHasResumableTrapHandler();
```

Comment:
As we handle both cases here, we can get rid of this `bool`.

Reply:
Going to let the current build complete then if it passes add a commit and squash to fix this up.

Reply:
Fixed in ff790c2